ARC-82: Migrate to access-control-service-api v3:

ahrytsiuk opened the pull request:
- Configure code generation
- Replace usages of v2 with v3

uesleilima commented on stream-dbs-clients/pom.xml:
```diff
@@ -234,6 +234,18 @@
                             <modelPackage>com.backbase.dbs.accesscontrol.api.integration.v2.model</modelPackage>
                         </configuration>
                     </execution>
+                    <execution>
```
perhaps we should remove from here the code generation for the v2 clients

ahrytsiuk replied:
@uesleilima do you know where `api.integration.v2.model` is used? I can't find any usages in the Stream itself. Is it safe to remove `generate-accesscontrol-integration-api-code` execution completely?

eltntawy commented:
Could you please verify testing the ingestion with [Backbase-bootstrap-task](https://github.com/baas-devops-reference/backbase-bootstrap-task) to make sure everything would work smoothly and will not break.

ahrytsiuk commented:
> Could you please verify testing the ingestion with [Backbase-bootstrap-task](https://github.com/baas-devops-reference/backbase-bootstrap-task) to make sure everything would work smoothly and will not break.

Hello @eltntawy I tested all changes with bootstrap task as well. It requires some minor updates (like update package names), but all other functionality related to legal entities works as expected.

ahrytsiuk commented:
Hello @Backbase/stream can we discuss failing "SonarCloud Code Analysis" check?

As you can see Sonar reports low code coverage on the "new code". However, it's better to say for "touched code".

In my PR I updated access control API from v2 to v3.

In v2 they had more segregated API clients, for example for legal entity - `legalEntityApi` and `legalEntitiesApi`. In v3 it's only `legalEntitiesApi`.
In v2, for service agreement they had - `serviceAgreementQueryApi`, `serviceAgreementApi` and `serviceAgreementsApi`. In v3 it's only `serviceAgreementsApi`.
And this list continues for other resources as well.

As you can see, they made a change in the spec which made a code generator to generate merged/squash version of API client.

And what I did - I just replaced calls from v2 to corresponding v3 class. Everything else is untouched. And this code already had a low coverage. 

@Backbase/stream, can you accept and merge this PR taking into account described above. Or sonar gate is not possible to by-pass?

eltntawy commented:
> Hello @Backbase/stream can we discuss failing "SonarCloud Code Analysis" check?
> 
> As you can see Sonar reports low code coverage on the "new code". However, it's better to say for "touched code".
> 
> In my PR I updated access control API from v2 to v3.
> 
> In v2 they had more segregated API clients, for example for legal entity - `legalEntityApi` and `legalEntitiesApi`. In v3 it's only `legalEntitiesApi`. In v2, for service agreement they had - `serviceAgreementQueryApi`, `serviceAgreementApi` and `serviceAgreementsApi`. In v3 it's only `serviceAgreementsApi`. And this list continues for other resources as well.
> 
> As you can see, they made a change in the spec which made a code generator to generate merged/squash version of API client.
> 
> And what I did - I just replaced calls from v2 to corresponding v3 class. Everything else is untouched. And this code already had a low coverage.
> 
> @Backbase/stream, can you accept and merge this PR taking into account described above. Or sonar gate is not possible to by-pass?

Understand the case here the overall code coverage is still 52%, we are working to increase the test coverage, so you can merge this PR.
